main.cc

```diff
@@ -1248,6 +1248,10 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 return sys_ks.start(snitch.local());
             }).get();
             cfg->host_id = sys_ks.local().load_local_host_id().get0();
+            messaging.invoke_on_all([host_id = cfg->host_id] (netw::messaging_service& ms) {
+                ms.set_my_host_id(host_id);
+            }).get();
+
```

Comment:
Is it possible to move this into `sys_ks.local().setup(snitch, messaging).get();` below the main code? All quirks to straightforward system-keyspace initialization are currently there and I'd like not to spread them around.

Reply:
The only important thing for `set_my_host_id` is that we call it before we start sending any message.

Reply:
Since host id is optional, we need an assert in establish-connection path that this key is set before the service is used to open outgoing connections.